### DISCUSSION TITLE
Host library in an external folder

### DISCUSSION THREAD

**Discussion by Asuryan34** [stashapp/stash#4346] [Q&A] [answered]
Hello.

Is it possible (If yes, how), to host all my library in a separate folder ? (Outside the stash app folder). I almost lost all my file when I had to reinstall the app (Didn't realised that all my library would be wiped.)

Note: I'm using Docker on a QNAP NAS.

**Comment by DogmaDragon** [comment, marked as answer]
Settings > Library > Add Directory. 

Just type the drive letter or mount point to start searching it.

**Reply by Asuryan34** [reply]
Hello.

Sorry, My question didn't include enough information :(
I'm using Docker on a QNAP NAS.

**Reply by DogmaDragon** [reply]
Stash accepts any mounted volume. See https://github.com/stashapp/stash/blob/develop/docker/production/README.md

**Reply by Tanarion40** [reply]
Good afternoon.
I have the same question but I can't figure out how to have Stash pointing to an external folder.
Currently, on my QNAP, I have few folders.

+Container
---application
------stash
---------data
------------scenes
+Scenes

So, instead of having stash pointing to /data/videos, I would like it to point to the Scenes folder higher in the hierarchy.
The "Settings > Library > Add Directory" doesn't allow me to go beyond the stash folder so I can't point to the right folder.

Any suggestion ?

(I understand that I should mount a volume before pointing to it, but how ?
